test/new-e2e/tests/apm/tests.go

```diff
@@ -68,6 +70,57 @@ func testTracesHaveContainerTag(t *testing.T, c *assert.CollectT, service string
 	assert.True(c, hasContainerTag(traces, fmt.Sprintf("container_name:%s", service)))
 }
 
+func testAutoVersionTraces(t *testing.T, c *assert.CollectT, intake *components.FakeIntake) {
+	t.Helper()
+	traces, err := intake.Client().GetTraces()
+	assert.NoError(c, err)
+	assert.NotEmpty(c, traces)
+	t.Log("Got traces", traces)
+	for _, tr := range traces {
+		for _, tp := range tr.TracerPayloads {
+			t.Log("Tracer Payload Tags:", tp.Tags["_dd.tags.container"])
+			ctags, ok := getContainerTags(t, tp)
+			assert.True(t, ok)
+			imageID, ok := ctags["image_id"]
+			assert.True(t, ok)
+			t.Logf("Got image ID: %v", imageID)
+			assert.Equal(t, "sha256:76909dd636e2ac542b2ff22b134e55861807ec3b88117d3d8573da5adb5b22d0", imageID)
+		}
+	}
+}
+
+func testAutoVersionStats(t *testing.T, c *assert.CollectT, intake *components.FakeIntake) {
+	t.Helper()
+	stats, err := intake.Client().GetAPMStats()
+	assert.NoError(c, err)
+	assert.NotEmpty(c, stats)
+	t.Log("Got apm stats:", spew.Sdump(stats))
+	for _, p := range stats {
+		for _, s := range p.StatsPayload.Stats {
+			t.Log("Client Payload:", spew.Sdump(s))
+			t.Logf("Got image ID: %v", s.GetImageTag())
+			assert.Equal(t, "sha256:76909dd636e2ac542b2ff22b134e55861807ec3b88117d3d8573da5adb5b22d0", s.GetImageTag())
```

Comment:
I'm a bit lost here - shouldn't https://github.com/DataDog/datadog-agent/pull/22828 add the image sha (as opposed to the image tag) ?